content/docs/userspace/gall-guide/8-subscriptions.md

```diff
@@ -0,0 +1,892 @@
+  $%  [%add =id =name]
+      [%del =id]
+      [%toggle =id]
+      [%rename =id =name]
+      [%initial =tasks]
+  ==
+--
+```
+
+This file defines most of the types for the agents. The list of to-do tasks will
+be stored in the state of the publisher agent as the `tasks` type, a `(map id task)`, where a `task` is a `[=name done=?]`. The set of ships allowed to
+subscribe will be stored in `friends`, a `(set @p)`, also in the publisher's
+state. After that, there are the head-tagged unions of accepted poke `action`s
+and `update`s for subscribers.
+
+#### `/mar/todo/action.hoon`
+
+```hoon
+/-  todo
+|_  =action:todo
+++  grab
+  |%
+  ++  noun  action:todo
+  --
+++  grow
+  |%
+  ++  noun  action
+  --
+++  grad  %noun
+--
+```
+
+This is a very simple mark file for the `action` type.
+
+#### `/mar/todo/update.hoon`
+
+```hoon
+/-  todo
+|_  =update:todo
+++  grab
+  |%
+  ++  noun  update:todo
+  --
+++  grow
+  |%
+  ++  noun  update
+  --
+++  grad  %noun
+--
+```
+
+This is a very simple mark file for the `update` type.
+
+### Publisher
+
+#### `/app/todo.hoon`
+
+```hoon
+/-  todo
+/+  default-agent, dbug
+|%
++$  versioned-state
+  $%  state-0
+  ==
++$  state-0  [%0 =friends:todo =tasks:todo]
++$  card  card:agent:gall
+--
+%-  agent:dbug
+=|  state-0
+=*  state  -
+^-  agent:gall
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %.n) bowl)
+::
+++  on-init
+  ^-  (quip card _this)
+  `this
+::
+++  on-save
+  ^-  vase
+  !>(state)
+::
+++  on-load
+  |=  old-state=vase
+  ^-  (quip card _this)
+  =/  old  !<(versioned-state old-state)
+  ?-  -.old
+    %0  `this(state old)
+  ==
+::
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  |^
+  ?>  =(src.bowl our.bowl)
+  ?+    mark  (on-poke:def mark vase)
+      %todo-action
+    =^  cards  state
+      (handle-poke !<(action:todo vase))
+    [cards this]
+  ==
+  ++  handle-poke
+    |=  =action:todo
+    ^-  (quip card _state)
+    ?-    -.action
+        %add
+      :_  state(tasks (~(put by tasks) now.bowl [name.action %.n]))
+      :~  :*  %give  %fact  ~[/updates]  %todo-update
+              !>(`update:todo`[%add now.bowl name.action])
+          ==
+      ==
+    ::
+         %del
+      :_  state(tasks (~(del by tasks) id.action))
+      :~  :*  %give  %fact  ~[/updates]  %todo-update
+              !>(`update:todo`action)
+          ==
+      ==
+    ::
+        %toggle
+      :_  %=  state
+            tasks  %+  ~(jab by tasks)
+                     id.action
+                   |=(=task:todo task(done !done.task))
+          ==
+      :~  :*  %give  %fact  ~[/updates]  %todo-update
+              !>(`update:todo`action)
+          ==
+      ==
+    ::
+        %rename
+      :_  %=  state
+            tasks  %+  ~(jab by tasks)
+                     id.action
+                   |=(=task:todo task(name name.action))
+          ==
+      :~  :*  %give  %fact  ~[/updates]  %todo-update
+              !>(`update:todo`action)
+          ==
+      ==
```

Comment:
```suggestion
        %toggle
      :-  :~  :*  %give  %fact  ~[/updates]  %todo-update
              !>(`update:todo`action)
          ==  ==
      %=  state
        tasks  %+  ~(jab by tasks)
                 id.action
               |=(=task:todo task(done !done.task))
      ==
    ::
        %rename
      :-  :~  :*  %give  %fact  ~[/updates]  %todo-update
              !>(`update:todo`action)
          ==  ==
      %=  state
        tasks  %+  ~(jab by tasks)
                 id.action
               |=(=task:todo task(name name.action))
      ==
```
I know I already rewrote this part once to get the lines under 80 characters, but now I realized this makes the state the heavier branch, so it ought to be on the bottom.